jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -1382,6 +1382,36 @@ public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingMode
                 _sessionHandler.setSessionTrackingModes(sessionTrackingModes);
         }
 
+        @Override
+        public int getSessionTimeout()
+        {
+            if (!isStarting())
+                throw new IllegalStateException();
+            if (!_enabled)
+                throw new UnsupportedOperationException();
+
+            int timeout = -1;
+            if (_sessionHandler != null)
+            {
+                int maxInactive = _sessionHandler.getMaxInactiveInterval();
+                timeout = (maxInactive < 0 ? 0 : maxInactive / 60);
+            }
+
+            return timeout;
+        }
+
+        @Override
+        public void setSessionTimeout(int sessionTimeout)
+        {
+            if (!isStarting())
+                throw new IllegalStateException();
+            if (!_enabled)
+                throw new UnsupportedOperationException();
+
+            if (_sessionHandler != null)
+                _sessionHandler.setMaxInactiveInterval((sessionTimeout < 0 ? 0 : sessionTimeout * 60));
+        }
```

Comment:
Just a thought that if I call `setSessionTimeout(-1)` and then `getSessionTimeout()` I get back `0` even if I set `-1`. Is that ok?

Reply:
I've made some changes: you should not be able to set more than the max minutes, nor less than the min minutes. Please take another look.